File: legacy/Conditional_Routing/nodejs/device_type.js
const MOBILE_PREFIX = '/mobile';
const TABLET_PREFIX = '/tablet';
const MOBILE_REG = /phone/i;
const TABLET_REG = /pad/i;

exports.handler = (event, context, callback) => {
    let req = event.req;
    
    let prefix = '';
    let headers = req.headers;
    if (headers['user-agent']) {
        const ua = headers['user-agent'][0];
        if (MOBILE_REG.test(ua)) {
            prefix = MOBILE_PREFIX;
        } else if (TABLET_REG.test(ua)) {
            prefix = TABLET_PREFIX;
        }
    }

    req.uri = prefix + req.uri;

    callback(null, req);
};

File: Conditional_Routing/lua/device_type.lua
local _M = {}

local MOBILE_PREFIX = '/mobile'
local TABLET_PREFIX = '/tablet'
local MOBILE_REG = 'phone'
local TABLET_REG = 'pad'

function _M.handler(event)
    local request = event.request
    local headers = request.headers

    local prefix = ''
    if type(headers['user-agent']) == 'string' then
        if re.find(headers['user-agent'], MOBILE_REG, 'jio') then
            prefix = MOBILE_PREFIX
        elseif re.find(headers['user-agent'], TABLET_REG, 'jio') then
            prefix = TABLET_PREFIX
        end
    end

    request.uri = prefix .. request.uri
    return request
end

return _M
